specs/AddHostObjectToScriptWithOrigins.md

```diff
@@ -0,0 +1,275 @@
+     runtimeclass CoreWebView2Frame
+    {
+        /// The name of the iframe from the iframe html tag declaring it.
+        /// Calling this method fails if it is called after the iframe is deleted.
+        String Name { get; };
+
+        /// Raised when the iframe changes its window.name property
+        event Windows.Foundation.TypedEventHandler<CoreWebView2Frame, Object> NameChanged;
+        /// The FrameDeleted event is raised when the iframe corresponding to this CoreWebView2Frame
+        /// object is removed or the document containing that iframe is destroyed
+        event Windows.Foundation.TypedEventHandler<CoreWebView2Frame, Object> FrameDeleted;
+
+        /// Add the provided host object to script running in the iframe with the
+        /// specified name for the list of the specified origins. The host object
+        /// will be accessible for this iframe only if the iframe's origin during
+        /// access matches one of the origins which are passed. The provided origins
+        /// will be normalized before comparing to the origin of the document.
+        /// So the scheme name is made lower case, the host will be punycode decoded
+        /// as appropriate, default port values will be removed, and so on.
+        /// This means the origin's host may be punycode encoded or not and will match
+        /// regardless.
+        /// If the iframe is declared with no src attribute its origin is considered
+        /// the same as the origin of the parent document.
+        /// List of origins will be treated as following:
+        /// 1. empty list - call will fail and no host object will be added for the
+        /// iframe;
+        /// 2. list with origins - during access to host object from iframe the
+        /// origin will be checked that it belongs to this list;
+        /// 3. list with "*" element - host object will be available for iframe for
+        /// all origins. We suggest not to use this feature without understanding
+        /// security implications of giving access to host object from from iframes
+        /// with unknown origins.
+        /// Calling this method fails if it is called after the iframe is deleted.
+        /// \snippet ScenarioAddHostObject.cpp AddHostObjectToScriptWithOrigins
+        /// For more information about host objects navigate to [AddHostObjectToScript]
+        void AddHostObjectToScriptWithOrigins(String name, Object rawObject, String[] origins);
+        /// Remove the host object specified by the name so that it is no longer
+        /// accessible from JavaScript code in the iframe. While new access
+        /// attempts are denied, if the object is already obtained by JavaScript code
+        /// in the iframe, the JavaScript code continues to have access to that
+        /// object. Calling this method for a name that is already removed or was never
+        /// added fails. If the iframe is deleted this method will return fail also.
```

Comment:
What does "fail" mean? It should be a no-op to delete something that was never there, for example (not an exception).

Reply:
see above